int32 encoder_hip
int32 encoder_knee
int32 encoder_heel
int32 encoder_spring